backend/src/test/java/ch/puzzle/pcts/service/business/CertificateCalculationBusinessServiceTest.java

```diff
@@ -0,0 +1,196 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CertificateCalculationBusinessServiceTest {
+
+    private static final Long ID = 1L;
+
+    @Mock
+    private CertificateCalculationValidationService validationService;
+
+    @Mock
+    private CertificateCalculationPersistenceService persistenceService;
+
+    @InjectMocks
+    private CertificateCalculationBusinessService businessService;
+
+    @Test
+    @DisplayName("Should get certificate calculations by calculation id")
+    void shouldGetByCalculationId() {
+        CertificateCalculation cc = mock(CertificateCalculation.class);
+        when(persistenceService.getByCalculationId(ID)).thenReturn(List.of(cc));
+
+        List<CertificateCalculation> result = businessService.getByCalculationId(ID);
+
+        assertEquals(1, result.size());
+        verify(persistenceService).getByCalculationId(ID);
+    }
+
+    @Test
+    @DisplayName("Should get certificate calculations by certificate id")
+    void shouldGetByCertificateId() {
+        CertificateCalculation cc = mock(CertificateCalculation.class);
+        when(persistenceService.getByCertificateId(ID)).thenReturn(List.of(cc));
+
+        List<CertificateCalculation> result = businessService.getByCertificateId(ID);
+
+        assertEquals(1, result.size());
+        verify(persistenceService).getByCertificateId(ID);
+    }
+
+    @Test
+    @DisplayName("Should calculate certificate points correctly")
+    void shouldCalculateCertificatePoints() {
+        Calculation calculation = mock(Calculation.class);
+        when(calculation.getRole()).thenReturn(mock(ch.puzzle.pcts.model.role.Role.class));
+        when(calculation.getRole().getIsManagement()).thenReturn(false);
+
+        CertificateType type = mock(CertificateType.class);
+        when(type.getCertificateKind()).thenReturn(CertificateKind.CERTIFICATE);
+        when(type.getPoints()).thenReturn(BigDecimal.TEN);
+
+        Certificate certificate = mock(Certificate.class);
+        when(certificate.getCertificateType()).thenReturn(type);
+
+        CertificateCalculation cc = mock(CertificateCalculation.class);
+        when(cc.getCalculation()).thenReturn(calculation);
+        when(cc.getCertificate()).thenReturn(certificate);
+        when(persistenceService.getByCalculationId(calculation.getId())).thenReturn(List.of(cc));
+
+        BigDecimal result = businessService.getCertificatePoints(calculation.getId());
+
+        assertEquals(BigDecimal.TEN, result);
+    }
+
+    @Test
+    @DisplayName("Should return zero points for empty list")
+    void shouldReturnZeroPoints() {
+        BigDecimal result = businessService.getCertificatePoints(0L);
```

Comment:
bro was?
what is that method even doing it gets the points of a Long?
If you get the points for the certificate with that id pls improve naming